src/content/release/breaking-changes/navigator-complete-route.md

```diff
@@ -0,0 +1,57 @@
+---
+title: When a route is removed from the stack, associated futures must complete
+description: >
+  Before this breaking change, routes created by Navigator and awaited for
+  results could never complete if the route was removed instead of popped.
+---
+
+## Summary
+
+When routes are pushed, developers can await them to be notified when they are
+popped. However, this didn't work when they were removed because the associated
+future was never completed.
+
+## Context
+
+All Navigator methods that call `remove` had this issue. By using `complete`,
+the issue is properly resolved, allowing developers to pass a result.
+
+## Description of change
+
+All Navigator methods have been updated to no longer call `remove` but instead
+use `complete`. Context menus are now built from the `contextMenuBuilder`
+parameter.
+
+All methods that directly use `complete` now accept an optional `result`
+parameter to return it to the associated future. Other methods that indirectly
+use `remove` currently return `null`. In the future, we might extend these
+methods with an optional callback function to allow developers to handle pop
+logic in indirect scenarios (such as `removeUntil`).
+
+## Migration guide
```

Comment:
Could we provide code examples before and after?

Reply:
Done